DockerBuild/test-in-container-postgres

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+# test-in-container-postgres -- Tests with PostgreSQL container linked
+#
+# Uses: https://hub.docker.com/_/postgres/
+##
+set -euo pipefail
+. "$(dirname "$0")"/config.bash
+
+: ${DOCKER_PGNAME:=test-postgres.$$}
+: ${POSTGRES_PASSWORD:=$RANDOM$RANDOM}
+: ${POSTGRES_START_TIME:=4} # seconds
+
+# run a postgres container
+trap 'docker rm -f "$DOCKER_PGNAME"' EXIT
+set -x
+docker run --detach --name "$DOCKER_PGNAME" \
+    --env POSTGRES_USER="$USER" \
+    --env POSTGRES_PASSWORD="$POSTGRES_PASSWORD" \
+    postgres \
+    #
+
+# wait until postgres starts
+postgres-has-started() {
+    docker run --rm --name "$DOCKER_PGNAME.check" \
+        --link "$DOCKER_PGNAME" \
+        --env PGPASSWORD="$POSTGRES_PASSWORD" \
+        postgres \
+        psql -h "$DOCKER_PGNAME" -U "$USER" -l
+}
+sleep $POSTGRES_START_TIME  # give it some initial time to start
+while ! postgres-has-started; do sleep 1.$RANDOM; done
```

Comment:
Sure we should definitely switch to the right way like docker compose soon.  I didn't have chance to play with it yet and didn't want to complicate things without fully understanding it, which will take some time.  Btw Greenplum tests should be fixed and included in Travis first (it's not working yet as the clients are missing in the build image).